### PR TITLE
Chore/various improvements

### PR DESCRIPTION
**Pull request by jakeklassen**
Some various _improvements_ depending on who you ask maybe haha.

* Cleaned up some styling. There where inconsistencies all over the place so I added prettier.
* `.npmignore` to ignore the `src` code since `lib` is published.
* Removed `lib` from cache and now ignore it since it's only relevant to npm.
* Added `prepublish` script to run for npm release.

Some changes to the `index.js` file were just to take better advantage of a closure and avoid `let` where possible. I think more can be done but I just did a quick once over.

### CODE REVIEW

**Comment by jakeklassen** [comment]
Switched to `prepublishOnly` since `prepublish` is also run on local npm install.

**Comment by jakeklassen** [comment]
Drop node v7 to stick with LTS releases and address yarn incompatibility:

Travis error log:
```
warning You are using Node "7.10.1" which is not supported and may encounter bugs or unexpected behavior. Yarn supports the following semver range: "^4.8.0 || ^5.7.0 || ^6.2.2 || >=8.0.0"
```

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "semi": true,
+  "singleQuote": true,
+  "trailingComma": "none"
```
Comment (DanJFletcher) [inline .prettierrc]: trailing commas make for less changed lines in git which is why I use them.

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "semi": true,
```
Comment (DanJFletcher) [inline .prettierrc]: I prefer having semi set to false.

**Review by DanJFletcher — CHANGES_REQUESTED** [review]
This is awesome! There was some seriously much-needed clean up in this repo haha 😅

I do prefer no semi-colons and trailing commas so we'll just have to update the style rules for that. Otherwise, this looks great! Thanks for the cleanup 👍

**Comment by jakeklassen** [comment]
@DanJFletcher Made those stylistic changes you like :+1:

**Comment by DanJFletcher** [comment]
> @DanJFletcher Made those stylistic changes you like 👍

You the man!

**Pull request merged**